jni/src/main/java/com/horizen/sigproofnative/NaiveThresholdSigProof.java

```diff
@@ -40,6 +40,7 @@ private static native boolean nativeVerifyProof(BackwardTransfer[] btList,
                                       byte[] endEpochBlockHash, byte[] prevEndEpochBlockHash,
                                       FieldElement constant, long quality, byte[] proof, boolean checkProof, String verificationKeyPath, boolean checkVk);
 
+    // TODO: check type of `constant`. Why not a byte[]?
```

Comment:
FieldElement can be passed to the Rust-side without a need for intermediate serialization into bytes

Reply:
I would propose to use byte array representations everywhere.
All the objects like `constant`, `certDataHash`, `customFields` and others on the SDK side are kept and stored as a pure java byte arrays. And, for example, when we need to use `constant` to verify snark proof we need to deserialize it to Rust FieldElement object first, then pass it to verify the proof, then free the object to avoid memory leaks.
Such an approach costs a lot, because we emit 2 JNI calls: FE deserialization, proof verification.
Instead we may use only 1 JNI call by passing `constant` as is and deserializing it inside `nativeVerifyProof` method.

The main goal is to decrease number of JNI calls as much as possible. Here it's just a proposal for the further discussions.